Content.Server/NPC/Components/NPCImprintingOnSpawnBehaviourComponent.cs

```diff
@@ -0,0 +1,34 @@
+using Content.Shared.Whitelist;
+using Robust.Shared.Collections;
+
+namespace Content.Server.NPC.Components;
+/// <summary>
+/// A component that makes the entity friendly to nearby creatures it sees on init.
+/// </summary>
+[RegisterComponent]
+public sealed partial class NPCImprintingOnSpawnBehaviourComponent : Component
+{
+    /// <summary>
+    /// filter who can be a friend to this creature
+    /// </summary>
+    [DataField]
+    public EntityWhitelist? Whitelist;
+
+    /// <summary>
+    /// the radius in which all creatures will be marked as exceptions to attack
+    /// </summary>
+    [DataField]
+    public float SearchRadius = 3f;
```

Comment:
This seems like it duplicates the vision radius in the NPC blackboard

Reply:
Or perhaps not, the initial impriting radius can be different from the final NPC search radius. Your variable name and comment made it hard for me to understand what this is for.